### PR TITLE
Implemented login

### PR DESCRIPTION
**Pull request by khubo**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,19 +1,40 @@
...
-  response.end(username)
+app.post('/user', (req, res) => {
+  let { username } = req.body
+
+  if(!username) {
+    res.statusCode = 400
+    return res.send({success: false, message: 'invalid username'})
+  }
+  Promise.coroutine(function * () {
+    let nameExists = yield db.checkUser(username)
```
Comment (einsteinreloaded) [inline src/index.js:28]: are we keeping username as unique.

**Review by einsteinreloaded — CHANGES_REQUESTED** [review]

File: src/public/bundle.js [hunk 2]
```diff
@@ -11,23 +10,28 @@ function UserAction (username) {
...
   }).then(
     response => response.text()
-)
+  )
 }
 
 function handleError () {
-  alert('Game Initialisation Failed!! Please try again')
+  console.log('Game Initialisation Failed!! Please try again')
 }
```
Comment (einsteinreloaded) [inline src/public/bundle.js:23]: i have made these changes already

**Review by einsteinreloaded — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (khubo) [reply src/index.js:28]: if it already exists, it would return something othrhwise null.

File: src/public/bundle.js (hunk 2 above)
Comment (khubo) [reply src/public/bundle.js:23]: i squashed teh commit and made some midfications

Comment (einsteinreloaded) [reply src/public/bundle.js:23]: https://github.com/einsteinreloaded/StoneAge/commit/b5a5ea500378181cc038d612cf718f423d6ac00a
check this some changes are already there.

File: src/index.js (hunk 1 above)
Comment (einsteinreloaded) [reply src/index.js:28]: i got that. I was asking can't we keep userid or sessionid as unique

**Review by einsteinreloaded — APPROVED** [review]

**Pull request merged**